src/drivers/MongoDriver.ts

```diff
@@ -1385,19 +1391,58 @@ export class MongoDriver implements DataStore {
   }): Promise<LearningObject> {
     const object = await this.db
       .collection<LearningObjectDocument>(COLLECTIONS.LEARNING_OBJECTS)
-      .findOne({ _id: params.id });
-    if (object) {
-      const author = await this.fetchUser(object.authorID);
-      return this.generateLearningObject(author, object, params.full);
+      .aggregate([
+        {
+          // match learning object by params.id
+          $match: { _id: params.id },
+        },
+        { $unwind: '$materials.files' },
```

Comment:
To avoid having to make a second query in the case were `materials.files` is an empty array or null, use the `preserveNullAndEmptyArrays` option for the `$unwind` operator.
https://docs.mongodb.com/manual/reference/operator/aggregation/unwind/

```suggestion
        { $unwind: {path: '$materials.files', preserveNullAndEmptyArrays: true} },
```